### PR TITLE
feat: allow defining security context on ingress controller helm chart

### PR DESCRIPTION
**Pull request by dgff07**
Allow defining the security context on apisix-ingress-controller as we have for apisix-dashboard and apisix.

### CODE REVIEW

File: charts/apisix-ingress-controller/templates/deployment.yaml [hunk 1]
```diff
@@ -39,6 +39,8 @@ spec:
         {{- toYaml . | nindent 8 }}
       {{- end }}
       serviceAccountName: {{ include "apisix-ingress-controller.serviceAccountName" . }}
+      securityContext:
```
Comment (tokers) [inline charts/apisix-ingress-controller/templates/deployment.yaml:42]: Should add docs in README.

Comment (dgff07) [reply charts/apisix-ingress-controller/templates/deployment.yaml:42]: @tokers 
I added the docs on README as requested

**Comment by tao12345666333** [comment]
LGTM

**Review by tokers — APPROVED** [review]

**Comment by dgff07** [comment]
I fixed an indentation problem on the new section added for the security context deployment.yaml file.

**Review by lingsamuel — APPROVED** [review]

**Review by tao12345666333 — APPROVED** [review]

**Comment by lijing-21** [comment]
Hi @dgff07 , thank you for your contribution!

Here is the Contributor T-shirt form[1], if you're interested, kindly take a look :)

[1] https://github.com/apache/apisix/blob/master/CONTRIBUTING.md#contributor-t-shirt

**Pull request merged**